src/main/java/org/ibp/api/java/impl/middleware/common/validator/LocationValidator.java

```diff
@@ -71,12 +99,171 @@ public void validateLocation(final BindingResult errors, final Integer locationI
+		final boolean isLocationUsedInNames = this.germplasmNameService.isLocationUsedInGermplasmName(locationId);
+		if (isLocationUsedInNames) {
+			this.errors.reject("location.is.used.in.names", "");
+			throw new ApiRequestValidationException(this.errors.getAllErrors());
+		}
+	}
+
+	private void validateLocationNotUsedInAttribute(final Integer locationId) {
+		final boolean isLocationUsedInAttributes = this.germplasmAttributeService.isLocationUsedInAttribute(locationId);
+		if (isLocationUsedInAttributes) {
+			this.errors.reject("location.is.used.in.attributes", "");
+			throw new ApiRequestValidationException(this.errors.getAllErrors());
+		}
+	}
+
+	private void validateLocationNotUsedInGermplasm(final Integer locationId) {
+		final boolean isLocationUsedInGermplasm = this.germplasmService.isLocationUsedInGermplasm(locationId);
+		if (isLocationUsedInGermplasm) {
+			this.errors.reject("location.is.used.in.germplasms", "");
+			throw new ApiRequestValidationException(this.errors.getAllErrors());
+		}
+	}
+
+	private void validateLocationNotUsedInStudy(final Integer locationId) {
+		final boolean isLocationUsedInStudies = this.studyService.isLocationUsedInStudy(locationId);
+		if (isLocationUsedInStudies) {
+			this.errors.reject("location.is.used.in.studies", "");
+			throw new ApiRequestValidationException(this.errors.getAllErrors());
+		}
+	}
+
+	private void validateExistingLocationId(final Integer locationId) {
+		this.validateLocation(this.errors, locationId);
+	}
+
+	private void validateLocationAbbr(final Integer locationId, final String locationAbbr) {
+		final LocationSearchRequest locationSearchRequest = new LocationSearchRequest();
+		locationSearchRequest.setLocationAbbreviations(Arrays.asList(locationAbbr));
+		final List<org.generationcp.middleware.api.location.Location> locationList =
+			this.locationService.getLocations(locationSearchRequest, null);
+
+		if (!locationList.isEmpty()) {
+			final List<String> locationAbbrIds = locationList.stream() //
+				.map(org.generationcp.middleware.api.location.Location::getLocationDbId) //
+				.collect(Collectors.toList());
+			locationAbbrIds.removeAll(Arrays.asList(locationId.toString()));
+			if (!locationAbbrIds.isEmpty()) {
+				this.errors.reject("location.abbr.is.in.used", "");
+				throw new ApiRequestValidationException(this.errors.getAllErrors());
+			}
+		}
+	}
+
+	private void validateLocationAbbr(final String locationAbbr) {
+		final LocationSearchRequest locationSearchRequest = new LocationSearchRequest();
+		locationSearchRequest.setLocationAbbreviations(Arrays.asList(locationAbbr));
+		final List<org.generationcp.middleware.api.location.Location> locationList =
+			this.locationService.getLocations(locationSearchRequest, null);
+
+		if (!locationList.isEmpty()) {
+			this.errors.reject("location.abbr.is.in.used", "");
+			throw new ApiRequestValidationException(this.errors.getAllErrors());
+		}
+	}
+
+	private void validateLocationType(final Integer locationTypeId) {
```

Comment:
both validateLocationType can be merged in one, if not reuse internal code and rename functions